src/backend/distributed/commands/utility_hook.c

```diff
@@ -468,6 +468,18 @@ ProcessUtilityInternal(PlannedStmt *pstmt,
 		PreprocessTruncateStatement((TruncateStmt *) parsetree);
 	}
 
+	if (IsA(parsetree, LockStmt))
+	{
+		LockStmt *stmt = (LockStmt *) parsetree;
+
+		ereport(NOTICE, errmsg("Processing LOCK command."));
+
+		ErrorIfUnsupportedLockStmt(stmt);
+		uint32 nowaitFlag = stmt->nowait ? DIST_LOCK_NOWAIT : 0;
+		AcquireDistributedLockOnRelations(stmt->relations, stmt->mode,
+										  DIST_LOCK_VIEWS_RECUR | nowaitFlag);
```

Comment:
Hmz, doesn't AcquireDistributedLockOnRelations take a list of relation OIDs?

stmt->relations in a list of RangeVars as far as I can tell.

Reply:
I changed the outside function's signature so it expects a list of RangeVars now. Also introduced a `AcquireDistributedLockOnRelations_Internal`.
This was done so that different statements that need to acquire locks, can just pass their `stmt->relations` to the function without doing any other work than specifying the function configurations with the flags I added.